### PR TITLE
Highlighter Updates

### PR DESCRIPTION
**Pull request by katjuell**
### What should this PR do?
Resolves [DEVED-243](https://sourcegraph.atlassian.net/browse/DEVED-243) by updating our highlighting logic and naming conventions across the relevant components.

Resolves the original issue, which stemmed from the fact that our internal regexp highlighting was not lining up exactly with the way that `prism` and `prism-react-renderer` break on language syntax. Ideally, we want to be able to use language syntax highlighting, while also adding another layer on top of it for author added highlights.

The proposed changes will allow this. Instead of `Highlighter` (Prism highlighting) and `OutputHighlighter` components, we have a shared `Highlighter` component that can be used for output blocks, and which allows user input for matching, as well as a `PrismSyntaxHighlighter` component, which does language syntax highlighting, before passing text content to the `Highlighter` component, where we can scan & search for the `matcher` prop.

This also updates the `Highlighter` logic to work with multiple, comma-separated matchers (please see `How to troubleshoot the Linux Permission denied error` for examples).

### Why are we making this change?
- We want language syntax highlighting to work, along with author added highlighting for emphasis.
- We want authors to have the ability to highlight as many terms as they want for emphasis.

### What are the acceptance criteria? 
- Language syntax highlighting should work correctly.
- Author added highlighting (via the `matcher` prop) should be correctly highlighted as a `mark`.
- `matcher` props with multiple, comma-separated terms should render correctly.

### How should this PR be tested?
- Go to any of the changed tutorials and check language syntax highlighting.
- Check that single `matcher` props are rendered correctly.
- Check that `matcher`s with multiple, comma-separated terms are rendered correctly (please see `How to troubleshoot the Linux Permission denied error` for examples).

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.

### CODE REVIEW

File: components/atoms/Highlighter/index.tsx [hunk 1]
```diff
@@ -1,56 +1,58 @@
...
+    StyledHighlighterMatch,
     StyledCodeWrapper,
-    StyledHighlighterWrapper 
+    StyledNonHighlighterMatch,
 } from './HighlighterStyles'
 
 interface Props {
     input: string
-    language: Language
+    prismSyntax?: boolean
```
Comment (katjuell) [inline components/atoms/Highlighter/index.tsx:13]: Basically, a way for us to preserve the `color` settings from #372 without also overriding styles from Prism.

File: README.md [hunk 2]
```diff
@@ -120,12 +120,12 @@ In the website code, the front-matter data is accessible as the [`frontMatter`](
 
 ### Code
 
-When working with code blocks, we use two components: `<Highlighter>` and `<OutputHighlighter>`. `<Highlighter>` includes language syntax highlighting and copy functionality, so we use this when we want to add code to a tutorial in a particular programming language (including `bash`). We can also pass a `matcher` prop to the component, to highlight particular parts of the code for emphasis (the emphasized code is styled as a `<mark>` element). 
+When working with code blocks, we use two components: `<PrismSyntaxHighlighter>` and `<OutputHighlighter>`. `<PrismSyntaxHighlighter>` includes language syntax highlighting and copy functionality, so we use this when we want to add code to a tutorial in a particular programming language (including `bash`). We can also pass a `matcher` prop to the component, to highlight particular parts of the code for emphasis (the emphasized code is styled as a `<mark>` element).
```
Comment (ltagliaferri) [inline README.md:123]: Should this be 

```
When working with code blocks, we use two components: 
`<PrismSyntaxHighlighter>` and `<Highlighter>`.
```

**Review by ltagliaferri — APPROVED** [review]
This is working as expected for me, thank you for digging into this. I had one comment about the terminology in the readme.

File: README.md (hunk 2 above)
Comment (katjuell) [reply README.md:123]: Oh yes! Thank you :)

**Pull request merged**